sklearn/feature_selection/_multivariate_selection.py

```diff
@@ -0,0 +1,214 @@
+    Returns
+    -------
+    stat : float that refers to the computed k_sample test statistic
+    
+    Notes
+    -----
+    
+    1) The k_sample testing problem can be thought of as a generalization of
+    the two sample testing problem. 
+    
+    2) By manipulating the inputs of the k_sample test, we create
+    concatenated versions of the inputs and a label matrix which are
+    paired. Then, any multivariate nonparametric test can be performed on
+    this data.
+    """
+    #extract data matrix of shape (_samples,_features) for each group
+    k_array = np.unique(y)
+    matrices = []
+    for i in k_array:
+        indices = np.where(y == i)[0] 
+        if len(X.shape) == 1:
+            xi = X[indices]
+        else:
+            xi = X[indices,:]
+        matrices.append(xi)
+    X = np.concatenate(matrices)
+    #one hot encode y for multivariate independence test
+    vs = []
+    for i in range(len(np.unique(y))):
+        n = matrices[i].shape[0]
+        encode = np.zeros(shape=(n, len(matrices)))
+        encode[:, i] = np.ones(shape=n)
+        vs.append(encode)
+    y = np.concatenate(vs)
+    
+    #mgc case
+    if score_func == "mgc":
+        warnings.filterwarnings("ignore")
+        mgc = multiscale_graphcorr(X,y,reps = 0)
+        stat = mgc.stat 
+    #default
+    else:
+        warnings.filterwarnings("ignore")
+        mgc = multiscale_graphcorr(X,y,reps = 0)
+        stat = mgc.stat 
+    return(stat)
+
+######################################################################
+# Transformer
+
+class MultivariateFeatureSelector(SelectorMixin, BaseEstimator):
+    """ Transformer that performs forward selection.
+    
+    This feature selector adds features (forward selection) to
+    form a feature subset. At each iteration, a parallel 
+    operation occurs in which a multivariate independence test 
+    is performed for each data matrix with the selected best 
+    features and an additional feature not yet selected. The 
+    additional feature associated with the highest multivariate
+    independence test statistic is then chosen as the next best 
+    feature. 
+    
+    Read more in the :ref:`User Guide <multivariate_feature_selection>`.
+    
+    Parameters
+    ----------
+    k: int, default=10
+        amount of features to select. 
+        
+    Attributes
+    ----------
+    features_ : array, shape (n_features,)
+        indices of all features in X
+    
+    best_features_ : array, shape (k,)
+         indices of selected k best features of features_
+         
+    Examples
+    --------
+    >>> from sklearn.datasets import load_digits
+    >>> from sklearn.feature_selection import MultivariateFeatureSelector
+    >>> X, y = load_digits(return_X_y=True)
+    >>> X.shape
+    (1797, 64)
+    >>> X_new = MultivariateFeatureSelector(k = 7).fit_transform(X, y)
+    >>> X_new.shape
+    (1797, 7)
+    """
+    
+    def __init__(self, k=10):
+        self.k = k
+        
+    class _Parallel:
```

Comment:
this doesn't have to be a nested class. I think it's confusing that way

Reply:
should have it's own docstring

Reply:
https://github.com/srecupe2/scikit-learn/commit/0cca1d0f8c89244981a30c22928271880153b171